libbeat/processors/annotate/kubernetes/podwatcher.go

```diff
@@ -0,0 +1,298 @@
+package kubernetes
+
+import (
+	"github.com/elastic/beats/libbeat/logp"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
```

Comment:
I think a custom client is not an option here, as this one is more advanced than a stateless REST consumer. Check SharedIndexInformer and Watch mechanism, the client code creates a copy of the state by watching k8s API